Use pipenv sync instead of pipenv install

nre-ableton opened the pull request:
It turns out that pipenv install is an alias for lock + sync. This
causes the lockfile to be updated, meaning we may not be installing
the packages which we expect.

---

Related to https://github.com/AbletonAppDev/devtools/issues/1715, ping @AbletonDevTools/gotham-city

nre-ableton committed:
Use pipenv sync instead of pipenv install

It turns out that pipenv install is an alias for lock + sync. This
causes the lockfile to be updated, meaning we may not be installing
the packages which we expect.

ala-ableton reviewed (approved):
LGTM :+1:

ala-ableton commented:
What about
https://github.com/AbletonDevTools/jenkins-node-scanner/blob/1440a6c2f87892ede17855d2432163a942e711eb/Dockerfile#L12
though?

ala-ableton commented:
There is also an occurrence of `pipenv install` in `README.md`.

nre-ableton committed:
Use --ignore-pipfile flag with pipenv install

This ensures that pipenv will not touch the lockfile during
installation. Normally we'd prefer to use pipfile sync, but there is
no --system option for that command (yet).

nre-ableton commented:
@ala-ableton Fixed!

ala-ableton commented on Dockerfile:
```diff
@@ -9,7 +9,7 @@ WORKDIR /jenkins_node_scanner
 COPY Pipfile.lock /jenkins_node_scanner
 RUN pip install --no-cache-dir pipenv
-RUN pipenv install --system
+RUN pipenv install --system --deploy
```
I don't understand this change. As mentioned in the previous commit, "pipenv install is an alias for lock + sync", so here we are doing lock (which might update the lockfile) and then `--deploy` will complain that the lockfile has changed? This means that we have to check that the lockfile is up-to-date before building this `Dockerfile`, i.e. we can never build this `Dockerfile` with the exact versions of packages that we want, only the latest... That doesn't sound right.

nre-ableton replied:
Yes, that is correct. However, I don't know of any other way to do this, because as mentioned, `pipenv sync` does not support the `--system` flag.

nre-ableton replied:
Also see: https://github.com/pypa/pipenv/issues/2227

nre-ableton commented:
@ala-ableton I think I came up with a better solution, mind having another look?

ala-ableton commented:
LGTM :+1:
